BaseTools/Plugin/OverrideValidation/OverrideValidation.py

```diff
@@ -596,22 +605,42 @@ def path_parse():
         if not os.path.isfile(Paths.TargetPath):
             raise RuntimeError("Module path is invalid.")
 
-
     if Paths.Version < 1 or Paths.Version > len(FORMAT_VERSIONS):
         raise RuntimeError("Version is invalid")
 
     if not os.path.isdir(Paths.WorkSpace):
         raise RuntimeError("Workspace path is invalid.")
-    if not os.path.isfile(Paths.TargetPath) and not os.path.isdir(Paths.TargetPath):
-        raise RuntimeError("Module path is invalid.")
-    # Needs to strip os.sep is to take care of the root path case
-    # For a folder, this will do nothing on a formatted abspath
-    # For a drive root, this will rip off the os.sep
-    if not os.path.normcase(Paths.TargetPath).startswith(os.path.normcase(Paths.WorkSpace.rstrip(os.sep)) + os.sep):
-        raise RuntimeError("Module is not within specified Workspace.")
+    if Paths.TargetPath is not None:
+        if not os.path.isfile(Paths.TargetPath) and not os.path.isdir(Paths.TargetPath):
+            raise RuntimeError("Module path is invalid.")
+        # Needs to strip os.sep is to take care of the root path case
+        # For a folder, this will do nothing on a formatted abspath
+        # For a drive root, this will rip off the os.sep
+        if not os.path.normcase(Paths.TargetPath).startswith(os.path.normcase(Paths.WorkSpace.rstrip(os.sep)) + os.sep):
+            raise RuntimeError("Module is not within specified Workspace.")
+
+    if Paths.RegenPath is not None:
+        if not os.path.isfile(Paths.RegenPath):
+            raise RuntimeError("Regen path is invalid.")
+        # Needs to strip os.sep is to take care of the root path case
+        # For a folder, this will do nothing on a formatted abspath
+        # For a drive root, this will rip off the os.sep
+        if not os.path.normcase(Paths.RegenPath).startswith(os.path.normcase(Paths.WorkSpace.rstrip(os.sep)) + os.sep):
+            raise RuntimeError("Module is not within specified Workspace.")
 
     return Paths
 
+# resolve relative paths to absolute path over the given list of prefixes.
+# prefixes: a list of path prefixes to check. if more than one prefix matches, those closer to the front of the list are preferred
+# relpath: a relative path to resolve against the prefixes.
+def resolvePath(prefixes, relpath):
```

Comment:
Same the comment below, if we could use the library method, we might be able to drop this implementation.